examples/widgets/annotated_cursor.py

```diff
@@ -0,0 +1,345 @@
+            # Position is transformed to pixel coordinates,
+            # an offset is added there and this is transformed back.
+            temp = [event.xdata, event.ydata]
+            temp = self.ax.transData.transform(temp)
+            temp = temp + self.offset
+            temp = self.ax.transData.inverted().transform(temp)
+            self.text.set_position(temp)
+            self.text.set_text(self.numberformat.format(*plotpoint))
+            self.text.set_visible(self.visible)
+
+            # Tell base class, that we have drawn something.
+            # Baseclass needs to know, that it needs to restore a clean
+            # backround, if the cursor leaves our figure context.
+            self.needclear = True
+
+            # Remember the recently drawn cursor position, so events for the
+            # same position (mouse moves slightly between two plot points)
+            # can be skipped
+            self.lastdrawnplotpoint = plotpoint
+        # otherwise, make text invisible
+        else:
+            self.text.set_visible(False)
+
+        # Draw changes. Canot use _update method of baseclass,
+        # because it would first restore the backround, which
+        # is done already and is not necessary.
+        if self.useblit:
+            self.ax.draw_artist(self.text)
+            self.canvas.blit(self.ax.bbox)
+        else:
+            # If blitting is deactivated, the overwritten _update call made
+            # by the base class immedeately returned.
+            # We still have to draw the changes.
+            self.canvas.draw_idle()
+
+    def setpos(self, xpos, ypos):
+        """
+        Finds the coordinates, which have to be shown in text.
+
+        The behaviour depends on the *dataaxis* attribute. Function looks
+        up the matching plot coordinate for the given mouse position.
+
+        Parameters
+        ----------
+        xpos : float
+            The current x position of the cursor in data coordinates.
+            Important if *dataaxis* is set to 'x'.
+        ypos : float
+            The current y position of the cursor in data coordinates.
+            Important if *dataaxis* is set to 'y'.
+
+        Returns
+        -------
+        ret : {2D array-like, None} The coordinates which should be displayed.
+            *None* is the fallback value.
+        """
+
+        # Get plot line data
+        xdata = self.line.get_xdata()
+        ydata = self.line.get_ydata()
+
+        # The dataaxis attribute decides, in which axis we look up which cursor
+        # coordinate.
+        if self.dataaxis == 'x':
+            pos = xpos
+            data = xdata
+            lim = self.ax.get_xlim()
+        elif self.dataaxis == 'y':
+            pos = ypos
+            data = ydata
+            lim = self.ax.get_ylim()
+        else:
+            raise ValueError(
+                "The data axis specifier {} should be 'x' or 'y'"
+                .format(self.dataaxis))
```

Comment:
```suggestion
            raise ValueError(f"The data axis specifier {self.dataaxis} should "
                             f"be 'x' or 'y'")
```

Reply:
I never knew that string format syntax, but I love it :D